### PR TITLE
cake-2301 - overwriting wiki variables api response

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/cake 
https://wikia-inc.atlassian.net/browse/CAKE-2301

Overwrite the wiki variables API response with relevant values from the CGS

### CODE REVIEW

File: extensions/wikia/FandomCreator/Hooks.php [hunk 1]
```diff
@@ -83,11 +73,74 @@ public static function onDesignSystemApiGetAllElements( WikiaDispatchableObject
...
-		if ( !empty( $theme->graphics->header ) ) {
-			$currentData['community-header']['background_image'] = $theme->graphics->header;
+		if ( !empty( $community->theme->graphics->header ) ) {
+			$data['community-header']['background_image'] = $community->theme->graphics->header;
+		}
+
+		// remove "explore" menu since it's forcefully added by the DS api. array_values is needed because array_filter
+		// turns the array into an assoc. array, so when json_encoded navigation is an object instead of an array
+		$data['community-header']['navigation'] = array_values(array_filter($data['community-header']['navigation'], function($navItem) {
+			$type = isset($navItem['type']) ? $navItem['type'] : false;
```
Comment (WikiaTech) [inline extensions/wikia/FandomCreator/Hooks.php]: MINOR Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

File: extensions/wikia/FandomCreator/Hooks.php [hunk 2]
```diff
@@ -83,11 +73,74 @@ public static function onDesignSystemApiGetAllElements( WikiaDispatchableObject
...
+			$titleType = isset($navItem['title']['type']) ? $navItem['title']['type'] : false;
```
Comment (WikiaTech) [inline extensions/wikia/FandomCreator/Hooks.php]: MINOR Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

File: extensions/wikia/FandomCreator/Hooks.php [hunk 3]
```diff
@@ -83,11 +73,74 @@ public static function onDesignSystemApiGetAllElements( WikiaDispatchableObject
...
+			$titleKey = isset($navItem['title']['key']) ? $navItem['title']['key'] : false;
```
Comment (WikiaTech) [inline extensions/wikia/FandomCreator/Hooks.php]: MINOR Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: extensions/wikia/FandomCreator/Hooks.php [hunk 4]
```diff
@@ -83,11 +73,74 @@ public static function onDesignSystemApiGetAllElements( WikiaDispatchableObject
...
-		if ( !empty( $theme->graphics->header ) ) {
-			$currentData['community-header']['background_image'] = $theme->graphics->header;
+		if ( !empty( $community->theme->graphics->header ) ) {
+			$data['community-header']['background_image'] = $community->theme->graphics->header;
+		}
+
+		// remove "explore" menu since it's forcefully added by the DS api. array_values is needed because array_filter
+		// turns the array into an assoc. array, so when json_encoded navigation is an object instead of an array
+		$data['community-header']['navigation'] = array_values( array_filter( $data['community-header']['navigation'], function( $navItem ) {
+			$type = isset( $navItem['type'] ) ? $navItem['type'] : false;
```
Comment (WikiaTech) [inline extensions/wikia/FandomCreator/Hooks.php]: MINOR Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

File: extensions/wikia/FandomCreator/Hooks.php [hunk 5]
```diff
@@ -83,11 +73,74 @@ public static function onDesignSystemApiGetAllElements( WikiaDispatchableObject
...
+			$titleType = isset( $navItem['title']['type'] ) ? $navItem['title']['type'] : false;
```
Comment (WikiaTech) [inline extensions/wikia/FandomCreator/Hooks.php]: MINOR Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

File: extensions/wikia/FandomCreator/Hooks.php [hunk 6]
```diff
@@ -83,11 +73,74 @@ public static function onDesignSystemApiGetAllElements( WikiaDispatchableObject
...
+			$titleKey = isset( $navItem['title']['key'] ) ? $navItem['title']['key'] : false;
```
Comment (WikiaTech) [inline extensions/wikia/FandomCreator/Hooks.php]: MINOR Remove the literal "false" boolean value. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1125)

**Comment by drsnyder** [comment]
LGTM

**Pull request merged**